www/lozd00/sp2/js/app.js

```diff
@@ -0,0 +1,420 @@
+        })
+        .then(
+            function(response) {
+                alert("TaskList successfully deleted.");
+                console.log("TaskList successfully deleted", response);
+
+                loadTaskLists();
+            },
+            function(err) {
+                alert("Error deleting TaskList: " + err.body);
+                console.error("Error deleting TaskList", err);
+            }
+        );
+    }
+}
+
+function formatDateToRFC3339(dateString) {
+    const date = new Date(dateString);
+    return date.toISOString();
+}
+
+var cityInfo = '';
+
+// Funkce pro ziskani informaci o meste
+function getCityInfo() {
+    return new Promise(function(resolve, reject) {
+      navigator.geolocation.getCurrentPosition(function(position) {
+        var lat = position.coords.latitude;
+        var lon = position.coords.longitude;
+      
+        const apiKey = '22c3f08e3b578b61e2cf6ceb689b4cba';
+        const apiUrl = `https://api.openweathermap.org/geo/1.0/reverse?lat=${lat}&lon=${lon}&limit=1&appid=${apiKey}&lang=cs`;
+      
+        fetch(apiUrl)
+          .then(function(response) {
+            // Zkontrolovat, zda je odpoved v poradku
+            if (response.ok) {
+              // Prevedeni odpovedi na JSON
+              return response.json();
+            } else {
+              // Zobrazeni chybove hlasky
+              console.log("OpenWeather API failed: " + response.status);
+              reject("OpenWeather API failed: " + response.status);
+            }
+          })
+          .then(function(data) {
+            // Prirazeni nazvu mesta a zeme
+            resolve(data[0].name + ", " + data[0].country);
+          });
+      });
+    });
+  }
+  
+   // Funkce pro pridani ukolu
+  function addTask() {
+      // Ziskani hodnot
+      var taskName = taskNameElement.value;
+      var taskDescription = taskDescriptionElement.value;
+      var taskDueDate = taskDueDateElement.value;
+      var taskPriority = taskPriorityElement.options[taskPriorityElement.selectedIndex].value;
+      var selectedListId = taskListElement.value;
+      var taskDueDateFormatted = formatDateToRFC3339(taskDueDate);
+  
+      // Kontrola, zda bylo zadano jmeno ukolu
+      if (!taskName) {
+          alert("Task name cannot be empty.");
+          return;
+      }
+  
+      // Kontrola, zda byl vybrany seznam ukolu
+      if (!selectedListId) {
+          alert("No task list selected.");
+          return;
+      }
+  
+      getCityInfo().then(function(cityInfo) {
+        // Volani API - kontrola duplikatu
+        gapi.client.tasks.tasks.list({
+            "tasklist": selectedListId
+        })
+        .then(function(response) {
+            var existingTasks = response.result.items;
+  
+            for (var i = 0; i < existingTasks.length; i++) {
+                if (existingTasks[i].title === taskName) {
+                    alert("Task with the same name already exists in the selected list.");

```

Comment:
neomezil bych to na unikatni nazvy tasku...

navic, kdyz tahle podminka je pravdiva, tak stejne zobrazujete, ze task was successfully created (line 411)